olive/engine/engine.py

```diff
@@ -82,22 +82,49 @@ def __init__(
         else:
             self.target = LocalSystem()
 
+        self.setup_accelerators(execution_providers)
+
+        # default evaluator
+        self.evaluator_config = None
+        if evaluator_config is not None:
+            self.evaluator_config = evaluator_config
+        elif self._config.evaluator is not None:
+            self.evaluator_config = self._config.evaluator
+
+        # dictionary of passes
+        self.pass_config = OrderedDict()
+
+        # {"pass_name": {"pass": pass, "host": host, "evaluator": evaluator, "clean_run_cache": clean_run_cache}}
+        self.passes = OrderedDict()
+
+        self.pass_flows = None
+        self.pass_flows_search_spaces = None
+
+        self.footprints = defaultdict(Footprint)
+
+        self.azureml_client_config = self._config.azureml_client_config
+
+        self._initialized = False
+
+    def setup_accelerators(self, execution_providers):
         if execution_providers is None:
             execution_providers = self._config.execution_providers
 
-        # verify the AzureML system have specified the execution providers
-        # Please note we could not use isinstance(target, AzureMLSystem) since it would import AzureML packages.
-        if self.target.system_type == SystemType.AzureML and execution_providers is None:
-            raise ValueError("AzureMLSystem requires execution providers to be specified.")
-        elif execution_providers is None:
-            if self.target.system_type in (SystemType.Local, SystemType.PythonEnvironment):
+        if not execution_providers:
+            if self.target.olive_managed_env:
+                raise ValueError("Managed environment requires execution providers to be specified.")
+            elif self.target.system_type == SystemType.AzureML:
+                # verify the AzureML system have specified the execution providers
+                # Please note we could not use isinstance(target, AzureMLSystem) since it would import AzureML packages.
+                raise ValueError("AzureMLSystem requires execution providers to be specified.")
+            elif self.target.system_type in (SystemType.Local, SystemType.PythonEnvironment):
                 execution_providers = self.target.get_supported_execution_providers()
-            else:
-                # for docker system and python system, we default use CPUExecutionProvider
+            elif self.target.system_type == SystemType.Docker:
+                # for docker system we default use CPUExecutionProvider
                 execution_providers = ["CPUExecutionProvider"]
```

Comment:
From my understanding, in this PR, you want to put off the logic to get EPs if the current system's ORT version unknow(like aml, docker).
Local: SystemType.Local, SystemType.PythonEnvironment
Remote: AML, Docker

Then if so, why we give the CPU eps for docker as default? Should we let user specify for docer system?